go/libkb/assertion.go

```diff
@@ -202,31 +232,67 @@ type AssertionHTTPS struct{ AssertionURLBase }
 type AssertionDNS struct{ AssertionURLBase }
 type AssertionFingerprint struct{ AssertionURLBase }
 
-func (b AssertionURLBase) Check() error {
+func (a AssertionHTTP) CheckAndNormalize() (AssertionURL, error) {
```

Comment:
tried that, but because of the mutation, i need to return an `AssertionURL` and an `AssertionURLBase` isn't an `AssertionURL`.